### PR TITLE
refactor(clusterMatch): tweak cluster match component args, export mo…

### PR DESCRIPTION
**Pull request by anotherchrisberry**
…re in lib

* simplifying API for the cluster matches component; also converting it to React and exposing it via the library
* fixing the color on btn-link hover (alto is light gray, which is barely visible)
* exposing new Spinner in library, tweaking old Spinner to make it clearer that it's legacy
* exposing taskReader and taskExecutor, no reason

### CODE REVIEW

File: app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx [hunk 1]
```diff
@@ -306,7 +306,7 @@ export class CreatePipelineModal extends React.Component<ICreatePipelineModalPro
           <Modal.Title>Create New {this.state.command.strategy ? 'Strategy' : 'Pipeline'}</Modal.Title>
         </Modal.Header>
         {this.state.loading && (
-          <Modal.Body style={{height: '200px'}}><Spinner radius={25} width={6} length={16} /></Modal.Body>
+          <Modal.Body style={{height: '200px'}}><LegacySpinner radius={25} width={6} length={16} /></Modal.Body>
```
Comment (archana-s) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: If you want to use the new spinners: 
React: <Spinner size="nano" />
Angular: <loading-spinner size="'nano'"></loading-spinner>

File: app/scripts/modules/core/src/delivery/create/TemplateDescription.tsx [hunk 2]
```diff
@@ -14,12 +14,12 @@ export interface ITemplateDescriptionProps {
...
 export class TemplateDescription extends React.Component<ITemplateDescriptionProps> {
   public render() {
-    const { Spinner } = NgReact;
+    const { LegacySpinner } = NgReact;
     return (
       <div className="col-md-12 template-description">
         {this.props.loading && (
           <div className="spinner">
-            <Spinner radius={5} width={3} length={8} />
+            <LegacySpinner radius={5} width={3} length={8} />
```
Comment (archana-s) [inline app/scripts/modules/core/src/delivery/create/TemplateDescription.tsx]: Potential for new spinner. It is available in react too

React: <Spinner size="nano" />
Angular: <loading-spinner size="'nano'"></loading-spinner>

File: app/scripts/modules/core/src/loadBalancer/LoadBalancers.tsx [hunk 3]
```diff
@@ -150,7 +150,7 @@ export class LoadBalancers extends React.Component<ILoadBalancersProps, ILoadBal
       </div>
     ) : (
       <div>
-        <h3><Spinner radius={30} width={8} length={16}/></h3>
+        <h3><LegacySpinner radius={30} width={8} length={16}/></h3>
```
Comment (archana-s) [inline app/scripts/modules/core/src/loadBalancer/LoadBalancers.tsx]: Same.

File: app/scripts/modules/core/src/presentation/main.less [hunk 4]
```diff
@@ -62,7 +62,7 @@ html {
     }
     color: var(--color-accent);
     &:active, &:hover, &:focus, &.active {
-      color: var(--color-alto);
+      color: var(--color-primary);
```
Comment (archana-s) [inline app/scripts/modules/core/src/presentation/main.less]: sorry :(

File: app/scripts/modules/core/src/widgets/cluster/ClusterMatches.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+export class ClusterMatches extends React.Component<IClusterMatchesProps> {
+
+  public render() {
+    const { matches } = this.props;
+    const { AccountTag } = NgReact;
+    if (!matches || !matches.length) {
+      return (<div>(no matches)</div>);
+    }
+    return (
+      <div>
```
Comment (archana-s) [inline app/scripts/modules/core/src/widgets/cluster/ClusterMatches.tsx]: ul li ?

Comment (archana-s) [reply app/scripts/modules/core/src/widgets/cluster/ClusterMatches.tsx]: .nostyle on ul removes all paddings/margins and list styles.

File: app/scripts/modules/core/src/widgets/cluster/ClusterMatches.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        {matches.map((match: IClusterMatch, index: number) => (
+          <div key={index}>
+            <AccountTag account={match.account}/>
+            <span className="break-word">{match.name}</span>
+            {match.regions && match.regions.length && <i> in {match.regions.join(', ')}</i>}
```
Comment (archana-s) [inline app/scripts/modules/core/src/widgets/cluster/ClusterMatches.tsx]: Do we use i for italics and icons?

Comment (anotherchrisberry) [reply app/scripts/modules/core/src/widgets/cluster/ClusterMatches.tsx]: We don't do this consistently. You'll see a lot of `<i>`s and `<em>`s throughout the codebase.

(I never connected `i` with `icon` until you mentioned it above, but it explains why font awesome uses it)

File: app/scripts/modules/core/src/widgets/cluster/ClusterMatches.tsx (hunk 5 above)
Comment (anotherchrisberry) [reply app/scripts/modules/core/src/widgets/cluster/ClusterMatches.tsx]: Makes more sense - will change

File: app/scripts/modules/core/src/presentation/main.less (hunk 4 above)
Comment (anotherchrisberry) [reply app/scripts/modules/core/src/presentation/main.less]: No worries - we knew there would be some little ones like this.

File: app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx (hunk 1 above)
Comment (anotherchrisberry) [reply app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: @archana-s I will leave it for you to convert these to the new spinners. I really just wanted to get the new React Spinner available in the core library build and make it more clear that these are the old spinners and we shouldn't use them.

**Comment by jrsquared** [comment]
lgtm

**Pull request merged**